common/src/main/java/org/dash/wallet/common/ui/enter_amount/AmountView.kt

```diff
@@ -104,6 +108,10 @@ class AmountView(context: Context, attrs: AttributeSet) : ConstraintLayout(conte
         updateCurrency()
         binding.convertDirectionBtn.setOnClickListener {
             dashToFiat = !dashToFiat
+            analytics.logEvent(if (dashToFiat) AnalyticsConstants.Coinbase.ENTER_AMOUNT_DASH
+                else AnalyticsConstants.Coinbase.ENTER_AMOUNT_FIAT,
+                bundleOf()
+            )
```

Comment:
Since this is a common component, then would not this log all currency direction changes as being from the coinbase module?

Reply:
Agree, this is related to my previous comment. The fact that we cannot easily inject AnalyticsService into this component is the first indication that logs are made at the wrong level. DashToFiat should be propagated through the viewModel into the coinbase and logged in there.

Reply:
Should that be the case for all of the component classes?  The Fragment that is using them and responding to events from the component should log the event, rather than the component itself.

Reply:
Yes, that should be the case. Common components should only be responsible for their own functionality and shouldn't know or care about event logging.